### DISCUSSION TITLE
MindsDB exploration

### DISCUSSION THREAD

**Discussion by Ishankoradia** [DalgoT4D/DDP_backend#441] [Ideas]
**<h2>Motivation & NGO use case</h2>**

One of the NGOs onboarded to Dalgo is using a whatsapp chatbot (Glific). Students on board the chatbot receive activities week-on-week  to work on. As we progress through the week, we see many students engaging less and eventually become dormant. These students don’t come back to the bot and are labeled as Churned out.
There are two levels where the students churn out

- Onboarding Churn: There is an initial drop in the user engagement during the onboarding stage. This means not all students reached initially are onboarded.

- Activity Churn: It has been observed that some percent of students Churn out from the total no of onboarded students. This could be due various reasons like content is too difficult, content is too easy or boring, personal/family issues, internet/network related issues etc.

Our POC focused on prediction of Activity Churn. Can we predict if a student is going to Churn or not accurately ? If yes, it allows TAP to take some kind of action to retain their “going to Churn out students”.

We set out to explore the tool [mindsdb](https://mindsdb.com/) for this 

**<h2>Our POC</h2>**

First approach using python scripts to train the model. We used `scikit-learn` to do this and tried various models like logistic regression, xgboost and random forest classifier. We wrote a fair bit of lines of code to achieve this. We got an accuracy of around ~5%.

We wanted to explore some tool that could do this basically an AI engine. We tried mindsdb and it worked pretty good. Some things I liked about mindsdb were
- We were able to train the model with a very few lines of code (~10). 
- Dont know yet if mindsdb does labelling and cleaning of data. Or if this is not possible. 
- One big advantage was creating different models and saving those after training for future use. In python scripts, it would be very difficult to tune and maintain the history of different models you tried with various parameters. 
- Also mindsdb directly connects to your db and can also push predictions to the db (need to check once).

@siddhant3030 can you add some pictures depicting different features of mindsdb
